hw13 Департамент ATM

Taisiya-Boon opened the pull request:
Написать приложение ATM Департамент:
          1) Департамент может содержать несколько ATM.
          2) Департамент может собирать сумму остатков со всех ATM.
          3) Департамент может инициировать событие – восстановить состояние всех
          ATM до начального (начальные состояния у разных ATM могут быть
          разными).

Taisiya-Boon committed:
hw13 Департамент ATM
          Написать приложение ATM Департамент:
          1) Департамент может содержать несколько ATM.
          2) Департамент может собирать сумму остатков со всех ATM.
          3) Департамент может инициировать событие – восстановить состояние всех
          ATM до начального (начальные состояния у разных ATM могут быть
          разными).

kataus commented on hw13-structural-patterns/src/main/java/ru/otus/ATMAsDepartmentElement.java:
```diff
@@ -0,0 +1,29 @@
+public class ATMAsDepartmentElement implements Listener {
+
+    @Getter
+    private ATMImpl atm;
```
Прям именно это реализация? А чем нас абстрактный контракт не устроил?

kataus commented on hw13-structural-patterns/src/main/java/ru/otus/ATMImpl.java:
```diff
@@ -0,0 +1,136 @@
+
+public class ATMImpl implements ATM {
+
+    private CellImpl[] cellMasses;
```
Я уже говорил про массивы в предыдущем ДЗ. Сейчас не приму, давайте на коллекции переберемся

kataus commented on hw13-structural-patterns/src/main/java/ru/otus/ATMImpl.java:
```diff
@@ -0,0 +1,136 @@
+    private ParBanknote[] parBanknoteInCell() {
+        ParBanknote[] par = new ParBanknote[cellMasses.length];
+        for (int i = 0; i < cellMasses.length; i++) {
+            switch (cellMasses[i].getParBanknote()) {
```
А почему бы на возвращать не номинал как значение, а как enum. И нам тогда не придется делать такой жуткий switch

kataus commented on hw13-structural-patterns/src/main/java/ru/otus/DepartmentATM.java:
```diff
@@ -0,0 +1,40 @@
+    @Override
+    public int sumTheResiduals() {
+        int sum = 0;
+        for(int i = 0; i < atms.size(); i++) {
```
for (Listner listener : atms) { - так гораздо проще обходить по списку

kataus commented on hw13-structural-patterns/src/main/java/ru/otus/Memento.java:
```diff
@@ -0,0 +1,15 @@
+
+public class Memento {
+
+    private ATM atm;
```
А вот тут уже корректно с интерфейсом и реализацией обошлись

Taisiya-Boon committed:
hw13 CellImpl возвращает Enum
     ATMImpl изменена реализация, массив cellMasses заменён листом
     DepartmentATM изменён способ обхода по списку  методах sumTheResiduals() и initialState()

Taisiya-Boon committed:
hw13 добавлен интерфейс к классу ATMAsDepartmentElement
